### ISSUE TITLE
[IDEA] escape character not identify correctly

### ISSUE THREAD

**Issue by daneshk** [ballerina-platform/ballerina-lang#13872]
**Description:**
Bellow code snippet showing an error in Idea IDE.
```ballerina
public final string SEPARATOR = (system:getenv("OS") == "") ? "/" : "\\";
```
Please check

**Comment by NipunaRanasinghe** [comment]
As per the offline discussion with @daneshk, this still occurs for top level definitions.

**Comment by NipunaRanasinghe** [comment]
Closing since fixed by #15196.
